### PR TITLE
Feat ✨(ui #8160) : Implement Analytics report page

### LINKED ISSUE #8160: Implement Analytics report page

### PR DESCRIPTION
**Pull request by Sachin-chaurasiya**
Closes #8160 
### Describe your changes :

I worked on the adding data insight page.

#
### Type of change :

- [x] New feature

#
### Frontend Preview (Screenshots) :

image

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers
@open-metadata/ui

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/src/components/DataInsightDetail/DescriptionInsight.tsx [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+          <ResponsiveContainer minHeight={400}>
+            <BarChart data={data} margin={BAR_CHART_MARGIN}>
+              <CartesianGrid strokeDasharray="3 3" />
+              <XAxis dataKey="timestamp" />
+
+              <YAxis />
+              <Tooltip />
+              <Legend
+                align="left"
+                content={(props) => renderLegend(props as LegendProps, `65.8%`)}
```
Comment (ShaileshParmar11) [inline openmetadata-ui/src/main/resources/ui/src/components/DataInsightDetail/DescriptionInsight.tsx:58]: why do we need 65.8% hard-coded as a string?

File: openmetadata-ui/src/main/resources/ui/src/components/DataInsightDetail/OwnerInsight.tsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+import { renderLegend } from '../../utils/DataInsightUtils';
+
+const OwnerInsight = () => {
+  const { data, entities } = getEntityOwnersData();
+
+  return (
+    <Card className="mt-4" data-testid="entity-summary-card-percentage">
+      <div data-testid="entity-summary-card-percentage-heder">
+        <Typography.Title level={5}>
+          Percentage of Entities With Owners
```
Comment (ShaileshParmar11) [inline openmetadata-ui/src/main/resources/ui/src/components/DataInsightDetail/OwnerInsight.tsx:42]: localaization

**Review by ShaileshParmar11 — COMMENTED** [review]
Lets add localization support as well wherever it is possible, also let few comments!

File: openmetadata-ui/src/main/resources/ui/src/components/DataInsightDetail/DescriptionInsight.tsx (hunk 1 above)
Comment (Sachin-chaurasiya) [reply openmetadata-ui/src/main/resources/ui/src/components/DataInsightDetail/DescriptionInsight.tsx:58]: We do not have API support, so we just added the mock data.

Comment (ShaileshParmar11) [reply openmetadata-ui/src/main/resources/ui/src/components/DataInsightDetail/DescriptionInsight.tsx:58]: ok got it!

**Comment by Sachin-chaurasiya** [comment]
Thanks, @ShaileshParmar11 , All comments are addressed.

File: openmetadata-ui/src/main/resources/ui/src/pages/DataInsightPage/DataInsightPage.component.tsx [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+              overlay={getMenuItems(TIER_FILTER, 'Tier.Tier1')}>
+              <Space>
+                Tier1
+                <DropDownIcon />
+              </Space>
+            </Dropdown>
+          </Space>
+        </Col>
+        <Col span={24}>
+          <DataInsightDetail />
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/pages/DataInsightPage/DataInsightPage.component.tsx:69]: Let's not have an overhead replace `<DataInsightDetail />` with below directly
```
<DataInsightSummary />
      <TotalEntityInsight />
      <DescriptionInsight />
      <OwnerInsight />
      <TierInsight />
      <TopViewEntities />
      <TopActiveUsers />
```
Also, components shouldn't be worrying about the layout. Page should be

File: openmetadata-ui/src/main/resources/ui/src/styles/app.less [hunk 4]
```diff
@@ -252,3 +252,17 @@
...
+.mr-2 {
+  margin-right: 0.5rem;
+}
+.mr-3 {
+  margin-right: 0.75rem;
+}
+
+.mr-4 {
+  margin-right: 1rem;
+}
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/styles/app.less:268]: space.less

File: openmetadata-ui/src/main/resources/ui/src/utils/DataInsightUtils.tsx [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+import { Menu, Typography } from 'antd';
+import { ItemType } from 'antd/lib/menu/hooks/useItems';
+import React from 'react';
+import { LegendProps, Surface } from 'recharts';
+
+export const getMenuItems = (items: ItemType[], defaultKey: string) => (
+  <Menu selectable defaultSelectedKeys={[defaultKey]} items={items} />
+);
+
+export const renderLegend = (props: LegendProps, total: string) => {
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/utils/DataInsightUtils.tsx:23]: If it's a component then it should be PascalCase and if it's a function we should rename props

File: openmetadata-ui/src/main/resources/ui/src/utils/TimeUtils.ts [hunk 6]
```diff
@@ -292,3 +292,12 @@ export const getFormattedDateFromSeconds = (
...
 ) => DateTime.fromSeconds(timeStamp || 0).toFormat(format || 'dd/MMM HH:mm');
+/**
+ * It takes a timestamp in milliseconds and returns a formatted date string
+ * @param {number} timeStamp - The timeStamp in milliseconds.
+ * @param {string} [format] - The format of the date you want to get default format is 'dd/MMM'.
+ */
+export const getFormattedDateFromMilliSeconds = (
+  timeStamp: number,
+  format?: string
+) => DateTime.fromMillis(timeStamp || 0).toFormat(format || 'dd/MMM');
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/utils/TimeUtils.ts:303]: Let's format using this and use it across the app.
@Sachin-chaurasiya Create an issue to update all the date across the application
https://moment.github.io/luxon/api-docs/index.html#datetimedatetime_med

**Review by chirag-madlani — CHANGES_REQUESTED** [review]

File: openmetadata-ui/src/main/resources/ui/src/pages/DataInsightPage/DataInsightPage.component.tsx (hunk 3 above)
Comment (Sachin-chaurasiya) [reply openmetadata-ui/src/main/resources/ui/src/pages/DataInsightPage/DataInsightPage.component.tsx:69]: Sure @chirag-madlani 
> Let's not have an overhead replace <DataInsightDetail /> with below directly.

@chirag-madlani , regarding this, the layout code is on the page only (DataInsightPage).
> Also, components shouldn't be worrying about the layout. Page should be

File: openmetadata-ui/src/main/resources/ui/src/utils/TimeUtils.ts (hunk 6 above)
Comment (Sachin-chaurasiya) [reply openmetadata-ui/src/main/resources/ui/src/utils/TimeUtils.ts:303]: @chirag-madlani , I did not get this part, can you suggest something?
> Let's format using this and use it across the app.

Comment (chirag-madlani) [reply openmetadata-ui/src/main/resources/ui/src/utils/TimeUtils.ts:303]: @Sachin-chaurasiya Did you check the link above?

Comment (Sachin-chaurasiya) [reply openmetadata-ui/src/main/resources/ui/src/utils/TimeUtils.ts:303]: Yes @chirag-madlani.

Comment (Sachin-chaurasiya) [reply openmetadata-ui/src/main/resources/ui/src/utils/TimeUtils.ts:303]: @chirag-madlani , You meant something like this?
```js
export const getFormattedDateFromMilliSeconds = (
  timeStamp: number,
) => DateTime.fromMillis(timeStamp || 0).toLocaleString(DateTime.DATETIME_MED);
```

**Comment by Sachin-chaurasiya** [comment]
Thanks, @chirag-madlani , All comments are addressed.

File: openmetadata-ui/src/main/resources/ui/src/components/DataInsightDetail/DataInsightTables.less [hunk 7]
```diff
@@ -0,0 +1,7 @@
+.data-insight-table-wrapper table {
+  border: none;
+  box-shadow: none;
+  .ant-table-thead tr > th {
+    color: #37352f80;
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/DataInsightDetail/DataInsightTables.less:5]: Why we are using a different colour for the table header? @devyani-kaushik

File: openmetadata-ui/src/main/resources/ui/src/components/DataInsightDetail/DataInsightSummary.tsx [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+import { OVERVIEW } from '../../pages/DataInsightPage/DataInsight.mock';
+
+const DataInsightSummary = () => {
+  const { t } = useTranslation();
+
+  return (
+    <Card data-testid="summary-card">
+      <div data-testid="summary-card-heder">
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/DataInsightDetail/DataInsightSummary.tsx:24]: Why do we need this extra div?
Can we use title prop of card instead?

**Review by chirag-madlani — CHANGES_REQUESTED** [review]

File: openmetadata-ui/src/main/resources/ui/src/components/DataInsightDetail/TopActiveUsers.tsx [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+  const { t } = useTranslation();
+
+  const columns: ColumnsType<ActiveUserView> = useMemo(
+    () => [
+      {
+        title: 'User',
+        dataIndex: 'userName',
+        key: 'userName',
+        render: (_, record) => (
+          <Typography.Text>{record.userName}</Typography.Text>
```
[the remark this answers is not in the document]
Comment (Sachin-chaurasiya) [reply openmetadata-ui/src/main/resources/ui/src/components/DataInsightDetail/TopActiveUsers.tsx:40]: My bad @chirag-madlani, I did not understand the `render` function first parameter properly, I will work on it and update it.

File: openmetadata-ui/src/main/resources/ui/src/components/DataInsightDetail/DescriptionInsight.tsx [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+import { getEntityDescriptionData } from '../../pages/DataInsightPage/DataInsight.mock';
+import { renderLegend } from '../../utils/DataInsightUtils';
+
+const DescriptionInsight = () => {
+  const { data, entities } = getEntityDescriptionData();
+  const { t } = useTranslation();
+
+  return (
+    <Card className="mt-4" data-testid="entity-summary-card-percentage">
+      <div data-testid="entity-summary-card-percentage-heder">
```
[the remark this answers is not in the document]
Comment (Sachin-chaurasiya) [reply openmetadata-ui/src/main/resources/ui/src/components/DataInsightDetail/DescriptionInsight.tsx:42]: Sure.

File: openmetadata-ui/src/main/resources/ui/src/components/DataInsightDetail/DataInsightSummary.tsx (hunk 8 above)
Comment (Sachin-chaurasiya) [reply openmetadata-ui/src/main/resources/ui/src/components/DataInsightDetail/DataInsightSummary.tsx:24]: Sure.

**Comment by Sachin-chaurasiya** [comment]
@chirag-madlani , All comments are addressed.

**Review by chirag-madlani — DISMISSED** [review]

File: openmetadata-ui/src/main/resources/ui/src/components/DataInsightDetail/TopActiveUsers.tsx [hunk 11]
```diff
@@ -0,0 +1,100 @@
...
+            {getDateTimeFromMilliSeconds(mostRecentSession)}
+          </Typography.Text>
+        ),
+      },
+      {
+        title: 'Total Sessions',
+        dataIndex: 'totalSessions',
+        key: 'totalSessions',
+        render: (totalSessions: number) => (
+          <Typography.Text>{totalSessions}</Typography.Text>
```
Comment (ShaileshParmar11) [inline openmetadata-ui/src/main/resources/ui/src/components/DataInsightDetail/TopActiveUsers.tsx:65]: Can we add localization support here as well?

File: openmetadata-ui/src/main/resources/ui/src/components/DataInsightDetail/TopViewEntities.tsx [hunk 12]
```diff
@@ -0,0 +1,109 @@
...
+      },
+      {
+        title: 'Tags',
+        dataIndex: 'tags',
+        key: 'tags',
+        render: (tags: string[]) => (
+          <Typography.Text>
+            {tags.map((tag, i) => (
+              <Tag key={i}>{tag}</Tag>
+            ))}
```
Comment (ShaileshParmar11) [inline openmetadata-ui/src/main/resources/ui/src/components/DataInsightDetail/TopViewEntities.tsx:57]: Can we add localization support here as well?

**Review by ShaileshParmar11 — COMMENTED** [review]
left few comment

**Review by ShaileshParmar11 — DISMISSED** [review]

**Review by ShaileshParmar11 — APPROVED** [review]

**Pull request merged**